src/util/ChartUtils.ts

```diff
@@ -976,33 +976,52 @@ export const getStackedData = (
   return stack(data);
 };
 
+type AxisId = string;
+type StackId = string | number | symbol;
+
+export type ParentStackGroup = {
+  hasStack: boolean;
+  stackGroups: Record<StackId, ChildStackGroup>;
+};
+
+export type ChildStackGroup = {
+  numericAxisId: string;
+  cateAxisId: string;
+  items: Array<ReactElement>;
+  stackedData?: ReadonlyArray<Series<Record<string, unknown>, string>>;
+};
+
+export type AxisStackGroups = Record<AxisId, ParentStackGroup>;
+
 export const getStackGroupsByAxisId = (
-  data: any,
-  _items: Array<any>,
+  data: ReadonlyArray<Record<string, unknown>> | undefined,
+  _items: Array<ReactElement>,
   numericAxisId: string,
   cateAxisId: string,
-  offsetType: any,
+  offsetType: StackOffsetType,
   reverseStackOrder: boolean,
-) => {
+): AxisStackGroups => {
   if (!data) {
     return null;
   }
 
   // reversing items to affect render order (for layering)
   const items = reverseStackOrder ? _items.reverse() : _items;
 
-  const stackGroups = items.reduce((result, item) => {
+  const parentStackGroupsInitialValue: Record<AxisId, ParentStackGroup> = {};
+
+  const stackGroups: Record<AxisId, ParentStackGroup> = items.reduce((result, item) => {
     const { stackId, hide } = item.props;
 
     if (hide) {
       return result;
     }
 
-    const axisId = item.props[numericAxisId];
-    const parentGroup = result[axisId] || { hasStack: false, stackGroups: {} };
+    const axisId: AxisId = item.props[numericAxisId];
```

Comment:
nit nit: does `AxisId` give any more clarity than if we just declared this as `string`. Imo I see `string` and I know its a string, I see a type alias of `AxisId` and I'm thinking its something more complex

Reply:
This area withs with maps of maps. I wanted to give names to the keys even though they are both `string`.

Before:
```
const obj: Record<string, Array>
const data: Record<string, Array>
```

After:
```
const obj: Record<AxisId, Array>
const data: Record<StackId, Array>
```

Reply:
Gotcha, reasoning makes sense in this specific case